src/physics_mg.cpp

```diff
@@ -28,12 +28,12 @@ void collision_mg(Particle& p)
   // Add to the collision counter for the particle
   p.n_collision()++;
 
-  if (settings::weight_window_checkpoint_collision)
-    apply_weight_windows(p);
-
   // Sample the reaction type
   sample_reaction(p);
 
+  if (settings::weight_window_checkpoint_collision)
+    apply_weight_windows(p);
+
```

Comment:
So now we are applying weight windows _after_ a sampled reaction. What effect does this have in practice?

Reply:
There is some discussion in this in the PR writeup in the "MG Weight Window Bug Fix" section. This bug is actually already addressed by #3202 -- @pshriwise is planning to get the tests for that PR finished and merged before this PR gets merged, so more discussion can be added to that PR if needed.